ts_src/transaction.ts

```diff
@@ -269,6 +288,73 @@ export class Transaction {
     );
   }
 
+  addIssuance(args: AddIssuanceArgs, inputIndex?: number): void {
+    // check the amounts.
+    if (args.assetAmount <= 0)
+      throw new Error('asset amount must be greater than zero.');
+    if (args.tokenAmount < 0) throw new Error('token amount must be positive.');
+
+    if (inputIndex) {
+      // check if the input exists
+      if (!this.ins[inputIndex])
+        throw new Error(`The input ${inputIndex} does not exist.`);
+      // check if the input is available for issuance.
+      if (this.ins[inputIndex].issuance)
+        throw new Error(`The input ${inputIndex} already has issuance data.`);
+    } else {
+      // verify if there is at least one input available.
+      if (this.ins.filter(i => !i.issuance).length === 0)
+        throw new Error(
+          'transaction needs at least one input without issuance data.',
+        );
+      // search and extract the input index.
+      inputIndex = this.ins.findIndex(i => !i.issuance);
+    }
+
+    const { hash, index } = this.ins[inputIndex];
+
+    // create an issuance object using the vout and the args
+    const issuance: Issuance = newIssuance(
+      args.assetAmount,
+      args.tokenAmount,
+      {
+        txHash: hash,
+        vout: index,
+      },
+      args.precision,
+      args.contract,
+    );
+
+    // add the issuance to the input.
+    this.ins[inputIndex].issuance = issuance;
+
+    const kOne = Buffer.from('01', 'hex');
+    const asset = Buffer.concat([kOne, calculateAsset(issuance.assetEntropy)]);
+    const assetScript = address.toOutputScript(args.assetAddress, args.net);
+
+    // send the asset amount to the asset address.
+    this.addOutput(
+      assetScript,
+      issuance.assetAmount,
+      asset,
+      Buffer.from('00', 'hex'),
+    );
+
+    const token = Buffer.concat([
+      kOne,
+      calculateReissuanceToken(issuance.assetEntropy, args.confidential),
+    ]);
+    const tokenScript = address.toOutputScript(args.tokenAddress, args.net);
+
+    // send the token amount to the token address.
+    this.addOutput(
+      tokenScript,
+      issuance.tokenAmount,
+      token,
+      Buffer.from('00', 'hex'),
+    );
+  }
```

Comment:
This output should be added only in case the provided tokenAmount is > 0.

The fact is that, in Liquid, one can issue an asset WITH or WITHOUT a reissuance token that allows the owner to re-mint that asset in the future.

For each amount defined in the issuance there must be a corresponding output in the transaction, hence if the token amount is not specified, the relative output must not be included in the transaction

Reply:
Fixed! Thank u